src/runners/cairoRunner.ts

```diff
@@ -47,17 +49,53 @@ export class CairoRunner {
   }
 
   // Run until the given PC is reached.
-  runUntilPc(finalPc: Relocatable, verbose: boolean = false): void {
-    while (this.vm.pc.offset < finalPc.offset) {
+  runUntilPc(
+    finalPc: Relocatable,
+    printMemory?: boolean,
+    printRelocatedMemory?: boolean
+  ): void {
+    while (!this.vm.pc.eq(finalPc)) {
       this.vm.step();
-      if (verbose) {
-        console.log(`AP: ${this.vm.ap.toString}`);
-        console.log(`FP: ${this.vm.fp.toString()}`);
-        console.log(`PC: ${this.vm.pc.toString()}`);
-
-        console.log(`[ap - 1]: ${this.vm.memory.get(this.vm.ap.sub(1))}`);
-        console.log(`[fp]: ${this.vm.memory.get(this.vm.fp)}`);
-      }
+      this.vm.relocate();
+    }
+    if (printMemory) {
+      console.log(this.vm.memory.toString());
+    }
+    if (printRelocatedMemory) {
+      console.log(this.vm.relocatedMemoryToString());
     }
   }
+
+  /** Export the trace little-endian encoded to a file */
+  exportTrace(filename: string = 'encoded_trace') {
+    const buffer = new ArrayBuffer((this.vm.relocatedTrace.length + 1) * 3 * 8);
+    const view = new DataView(buffer);
+
+    this.vm.relocatedTrace.forEach((entry, step) => {
+      const byteOffset = step * 3 * 8;
+      view.setBigUint64(byteOffset, entry.ap.toBigInt(), true);
+      view.setBigUint64(byteOffset + 8, entry.fp.toBigInt(), true);
+      view.setBigUint64(byteOffset + 2 * 8, entry.pc.toBigInt(), true);
+    });
+
+    fs.writeFile(filename, buffer, { flag: 'w+' }, (err) => {
+      if (err) throw err;
+    });
+  }
+
+  /** Export the relocated memory little-endian encoded to a file */
+  exportMemory(filename: string = 'encoded_memory') {
+    const buffer = new ArrayBuffer(this.vm.relocatedMemory.size * 5 * 8);
+    const view = new DataView(buffer);
+
+    this.vm.relocatedMemory.forEach((value, address) => {
+      const byteOffset = (address - 1) * 5 * 8;
```

Comment:
what is this `*5` ?

Reply:
The address is written on 1 byte while the value (being a Felt) is written on 4 bytes, thus the `* 5`